BuildForPublication.cmd

```diff
@@ -3,12 +3,21 @@ goto :main
 
 :main
 setlocal
+  set errorlevel=
+  set BuildConfiguration=Release
+  set VersionSuffix=build0047
+  set PackageVersion=1.0.0-alpha-%VersionSuffix%
 
-set PackageVersion=1.0.0-alpha-build0043
-set VersionSuffix=build0043
+  echo/==================
+  echo/ Building version %PackageVersion% NuGet packages.
+  echo/==================
 
-echo Building version %PackageVersion% NuGet packages.
+  set LocalDotNet_ToolsDir=%~dp0tools
+  if exist "%LocalDotNet_ToolsDir%" rmdir /s /q "%LocalDotNet_ToolsDir%"
```

Comment:
I would think it more reasonable here to require the user to explicitly perform the cleaning themselves.

That is to say, if there are untracked changes or if there are artifacts where there shouldn't be, then fail the script with an appropriate error message and let the user resolve it themselves.

Otherwise, just let the script always run and give a warning under the above scenarios (which I think is the even better case -- perhaps the user wants to test some things before committing, which is a completely reasonable scenario).